checks/sast.go

```diff
@@ -61,36 +62,39 @@ func SAST(c *checker.CheckRequest) checker.CheckResult {
 	if sastScore != checker.InconclusiveResultScore &&
 		codeQlScore != checker.InconclusiveResultScore {
 		switch {
-		// sastScore >= codeQlScore only happens if:
-		// - sastScore is maximum and codeQl is enabled OR
-		// - sastScore is minimum and codeQl is not enabled
-		case sastScore == checker.MaxResultScore && sastScore >= codeQlScore:
-			return checker.CreateProportionalScoreResult(CheckSAST,
-				"all commmits are checked with a SAST tool", sastScore, checker.MaxResultScore)
-		case sastScore != checker.MaxResultScore && sastScore >= codeQlScore:
-			return checker.CreateMinScoreResult(CheckSAST,
-				"no SAST tool detected")
+		case sastScore == checker.MaxResultScore:
+			return checker.CreateMaxScoreResult(CheckSAST, "a SAST tool is run on all commits")
+		case codeQlScore == checker.MinResultScore:
+			return checker.CreateResultWithScore(CheckSAST,
+				checker.NormalizeReason("no SAST is run on all commits", sastScore), sastScore)
+
 		// codeQl is enabled and sast has 0+ (but not all) PRs checks.
 		case codeQlScore == checker.MaxResultScore:
 			const sastWeight = 3
 			const codeQlWeight = 7
 			score := checker.AggregateScoresWithWeight(map[int]int{sastScore: sastWeight, codeQlScore: codeQlWeight})
-			return checker.CreateResultWithScore(CheckSAST, "SAST tool detected but not used on all commmits", score)
+			return checker.CreateResultWithScore(CheckSAST, "SAST tool detected but not run on all commmits", score)
 		default:
 			return checker.CreateRuntimeErrorResult(CheckSAST, sce.Create(sce.ErrScorecardInternal, "contact team"))
 		}
 	}
 
-	// CodeQl inconclusive.
-	// Can never happen currently.
+	// Sast inconclusive.
 	if codeQlScore != checker.InconclusiveResultScore {
-		return checker.CreateResultWithScore(CheckSAST, "internal error running codeQl test", codeQlScore)
+		if codeQlScore == checker.MaxResultScore {
+			return checker.CreateMaxScoreResult(CheckSAST, "SAST tool detected")
+		}
+		return checker.CreateMinScoreResult(CheckSAST, "no SAST tool detected")
 	}
 
-	// Sast inconclusive.
-	// Only happens if no merges are detected.
+	// CodeQl inconclusive.
 	if sastScore != checker.InconclusiveResultScore {
-		return checker.CreateResultWithScore(CheckSAST, "no merges detected", sastScore)
+		if sastScore == checker.MaxResultScore {
+			return checker.CreateMaxScoreResult(CheckSAST, "a SAST tool is run on all commits")
+		}
+
+		return checker.CreateResultWithScore(CheckSAST,
+			checker.NormalizeReason("no SAST is run on all commits", sastScore), sastScore)
```

Comment:
slightly confusing to read.
maybe "SAST tool is not run on all commits" ?

Reply:
done.